File: cs/Scenarios/Server.SimaticNodes/SimaticNodeIdFactory.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace SimaticNodes
{
    using Opc.UaFx;

    public class SimaticNodeIdFactory : OpcNominalNodeIdFactory
    {
        #region ---------- Public constructors ----------

        public SimaticNodeIdFactory()
            : base()
        {
            this.Separator = '.';
        }

        #endregion

        #region ---------- Protected methods ----------

        protected override OpcNodeId Create(
                OpcContext context,
                OpcNamespace nodeNamespace,
                IOpcNodeInfo node,
                OpcName nodeName)
        {
            if (!OpcName.IsNullOrEmpty(nodeName))
                nodeName = nodeName.Namespace.GetName("\"" + nodeName.Value + "\"");

            return base.Create(context, nodeNamespace, node, nodeName);
        }

        #endregion
    }
}

File: vb/Scenarios/Server.SimaticNodes/SimaticNodeIdFactory.vb
' Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports Opc.UaFx

Namespace SimaticNodes
    Public Class SimaticNodeIdFactory
        Inherits OpcNominalNodeIdFactory

        Public Sub New()
            MyBase.New()
            Me.Separator = "."c
        End Sub

        Protected Overrides Function Create(
                context As OpcContext,
                nodeNamespace As OpcNamespace,
                node As IOpcNodeInfo,
                nodeName As OpcName) As OpcNodeId
            If Not OpcName.IsNullOrEmpty(nodeName) Then
                nodeName = nodeName.Namespace.GetName("""" & nodeName.Value & """")
            End If

            Return MyBase.Create(context, nodeNamespace, node, nodeName)
        End Function
    End Class
End Namespace
